test/e2e/common.go

```diff
@@ -208,3 +209,18 @@ func WaitForBmhInPowerState(ctx context.Context, input WaitForBmhInPowerStateInp
 		g.Expect(bmh.Status.PoweredOn).To(Equal(input.State == PoweredOn))
 	}, intervals...).Should(Succeed())
 }
+
+func buildKustomizeManifest(ctx context.Context, source string) ([]byte, error) {
+	kustomizePath, ok := os.LookupEnv("BMO_KUSTOMIZE_PATH")
+	if !ok {
+		kustomizePath = "kustomize"
+	}
+	kustomize := exec.NewCommand(
+		exec.WithCommand(kustomizePath),
+		exec.WithArgs("build", source))
+	stdout, stderr, err := kustomize.Run(ctx)
+	if err != nil {
+		return []byte{}, errors.Wrapf(err, "failed to execute kustomize: %s", stderr)
+	}
+	return stdout, nil
+}
```

Comment:
This does not get rid of the `os.exec`, we just use it through a nicer wrapper. IMO this is not worth it. We still need the binary (kustomize instead of kubectl) and we still execute it from the go code.

Reply:
Thank you. Now I realize I chose this method simply because CAPI is using it (the exec wrapper is from them too), but I agree that it's not worth it. I'll change to using one of the packages.